src/controller/migrate/materialization/mod.rs

```diff
@@ -304,6 +304,10 @@ impl Materializations {
                 able = false;
             }
 
+            if graph[ni].is_internal() && graph[ni].must_materialize() {
```

Comment:
I'm confused by this. Is `must_materialize` used to indicate that *full* materialization is necessary?

Reply:
Yes. The idea is that trigger nodes should always be fully materialized, otherwise group universes wouldn't exist until a user universe that belongs to that group is created. This might be something that we want to do, but it has some downsides: 1) it makes user universe creation slower, since it needs to wait for the necessary groups to be creates and 2) it's not super easy to guarantee order between migrations (the group universe needs to be created before the the user universe)

Reply:
It also gets really awkward because you are now much more likely to end up with full-below-partial :/ If we do indeed need it though, the method should be renamed to `requires_full_materialization`, and there should be a comment in the impl for `Trigger` giving the rationale for why it has to be full.